### ISSUE TITLE
[JENKINS-18778] Builds disappear from build history

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#6801]
<p>We have 4-5 projects (out of ~40 projects) that all the builds disappear from the "Build History" from time to time.  There was no external event (like restarting Jenkins, or triggering a new build for those projects).<br/>
We know those builds are there from the Jobs directory, but they just don't show on the Build history.</p>

<p>Only after we restart Jenkins, and after restart the builds come to the build history.</p>

**Attachments:**

- [screenshot-1.jpg](https://issues.jenkins.io/secure/attachment/24004/screenshot-1.jpg)
- [screenshot-2.jpg](https://issues.jenkins.io/secure/attachment/24005/screenshot-2.jpg)
- [screenshot-3.jpg](https://issues.jenkins.io/secure/attachment/24006/screenshot-3.jpg)

---
<details><summary><i>Originally reported by <a title="Dominic Lam" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ctlam">ctlam</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-18778" target="_blank">Builds disappear from build history</a></i></summary>
<i><ul>
<li><b>reported by</b>: ctlam
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2014-05-18T12:44:41+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 5
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=prsgan">prsgan</a>:</i>
<p>The similar issue also noticed with version 1.509.2</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=psftw">psftw</a>:</i>
<p>I can confirm on 1.531 that this issue only seems to happen when I've renamed a job and have not restarted Jenkins yet.  After a restart, the job history returns.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=psftw">psftw</a>:</i>
<p>Downgrading to minor due to easy workaround and no loss of data.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=oleg_nenashev">oleg_nenashev</a>:</i>
<p>What about <a href="https://issues.jenkins.io/browse/JENKINS-18678" title="Builds disappear some time after renaming job" class="issue-link" data-issue-key="JENKINS-18678"><del>JENKINS-18678</del></a>?<br/>
Do you see the issue on newer versions?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Based on comment by requester, this is clearly a duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-18678" title="Builds disappear some time after renaming job" class="issue-link" data-issue-key="JENKINS-18678"><del>JENKINS-18678</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Duplicates: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-06-a2/issues?q=in%3Atitle%20JENKINS-18678">JENKINS-18678</a>]</i>
